# Skyte spurv med kanoner

A)
Til denne øvelsen skal vi annvende veiformel 2 for å beskrive bevegelsen til en kanonkule skutt ut med en sprettert. her ser vi at startfarten er representert og multipliert siden hastighet multiplisert med tid blir strekning. samtidig ser vi at $sin\theta$ er tatt med. på grunn av at startfarten fører til bevegelse i både x- og y- aksen vi må derfor annvende trigonometri for å forklare innvirkning på ballens y-akse (høyden). samtidig har vi også med en endrende hastighet som virker neddover hvilket er tyngde akselerasjonen multiplisert med tiden opphøyd i annen. dividert med to. grunnen til at dette leddet har negativ notasjon er at akselerasjonen virker neddover.
    
    
$ y(t) = v_0 * sin\theta*t -\frac{1}{2}gt^2 $
    

B) Her multipliserer vi startfart med tiden for ballens bane. slik finner vi strekning men siden startfarten ikke beveger seg loddrett må vi multiplisere med cosinus og bruke trigonometri for å finne den faktiske lengden.

C,D)  Bruker numerisk derivasjon og newtons metode for å finne tid og deretter styrekning.

In [28]:
from pylab import*
n = 100 
x = float(input("sett inn estimert tid i sekunder: "))
dx = 1E-8
def f(x):
    return 18* x * sin(pi/10) - 0.5 * 9.81 * x**2
def D(x):
    return (f(x + dx) -f(x))/dx   
for i in range(n):
    x = x - (f(x))/D(x)
    
a = 18*cos(pi/10) * x    

print(x, "sekunder")
print(a,"meter")

sett inn estimert tid i sekunder: 1
19.413090901402775 meter
1.1340073188071462 sekunder


In [31]:
c = (36*sin(pi/10))/9.81
print(c, "sekunder")

1.1340073188071464 sekunder


Her ser vi at annvendelse av newtons metode gir i svært stor nøyaktighet tiden ballen er i lufta.

E) 

In [37]:
n = 100
l = array([6, 5, 4, 3])
x = float(input("sett inn estimert tid i sekunder: "))
dx = 1E-8

def f(x):
    return 0.3 + 5.8 * x * sin(pi/l) - 0.5 * 9.81 * x**2
def D(x):
    return (f(x + dx) -f(x))/dx   
for i in range(n):
    x = x - (f(x))/D(x)
a = 5.8 * cos(pi/l) * x    

print(x, "sekunder")
print(a,"meter")

sett inn estimert tid i sekunder: 1
[0.68104028 0.77405205 0.90380231 1.08064415] sekunder
[3.42082947 3.63208332 3.7066915  3.13386803] meter


her deffinerer vi endringen av vinkelen i en array slik får vi tid og avstand for vinklene. da ser vi at vinkel 45 grader er mest ideel.

F)

In [38]:
n = 100
x = float(input("sett inn estimert tid i sekunder: "))
dx = 1E-8
def f(x):
    return 0.3 + 5.8 * x * sin(pi/4) - 0.5 * 9.81 * x**2
def D(x):
    return (f(x + dx) -f(x))/dx   
for i in range(n):
    x = x - (f(x))/D(x)
a = 5.8 * cos(pi/4) * x    

print(x, "sekunder")
print(a,"meter")

sett inn estimert tid i sekunder: 1
0.9038023091058767 sekunder
3.7066915014007877 meter


ved 45 grader målte vi 
en startfart på 5.79, 5.80, 5.87
og kulen tilbakela avstandene 3.75, 3.68, 3.63

her ser vi en klar ovenestemmelse mellom målinger og beregninger og noen av feilene kan være grunnet feilkilder.

Mulige feilkilder:
- vi har ikke tatt høyde for luftmotstand
- Feilmålinger
- unøyaktige vinkler på kanon